core/db/migrate/20210122110141_add_name_to_spree_addresses.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class AddNameToSpreeAddresses < ActiveRecord::Migration[5.2]
+  class AddressForMigration < ApplicationRecord
+    self.table_name = 'spree_addresses'
+  end
+
+  def up
+    if AddressForMigration.where.not(firstname: [nil, '']).exists?
+      AddNameToSpreeAddresses.handle_exisiting_data
+    end
```

Comment:
How should a store be able to add the `name` column if we stop the migration here when existing addresses exist? Move all names into the `lastname` column first?

I think adding the `name` column first should be fine.

Reply:
@tvdeyen this migration is intentionally stopped before the change for people that do have at least one record that needs manual attention. They can change the migration according to their context. We will provide some alternatives in following PRs.

Reply:
But adding a new column should be ok, no? The exception text talks about moving the data. Where should the data be moved to if there is no column? Maybe just print the message out after we introduced the column?

Reply:
Adding the new column and then raise will make the migration execute and rollback. That works but I can't see a lot of difference now. 

But yes, when we'll introduce some alternatives approaches for people to choose (as we did in [this other migration](https://github.com/solidusio/solidus/blob/bfc998da4712ad3b04372462439e0853b84cf9c3/core/db/migrate/20190106184413_remove_code_from_spree_promotions.rb#L30)), it will probably be better to have this check after the column addition so that we don't ask to also move the code around.

Reply:
@tvdeyen We removed the data migration helper code. It is better to provide an upgrade rake task that will handle this and other things needed for the major upgrade. We can expect broken apps if that task will not be executed but after all, that's what a major does, isn't it?

Reply:
Agreed 👍🏻